File: stack.java

public class stack {
    class Node{
        int data;
        Node next;

        Node(int data){
            this.data=data;
        }
    }
public Node top=null;
    public void push(int data) {
        Node newNode=new Node(data);
        if (top==null) {
            top=newNode;
        }
        else{
            newNode.next=top;
            top=newNode;
        }
    }

    public void pop() {
        top=top.next;
    }

    public void display() {
        Node current=top;
        if (top==null) {
            System.out.println("stack is empty");
        }
        while(current!=null){
            System.out.println(current.data);
            current=current.next;
        }
    }
    
    public static void main(String[] args) {
        stack list=new stack();
        list.push(10);
        list.push(25);
        list.push(45);
        list.push(89);
        list.display();
        list.pop();
        System.out.println("pop");
        list.display();
    }
}

File: stack.js
class Node{
    constructor(data){
        this.data=data;
        this.next=null

    }
}

class stack{
    constructor(){
        this.top=null;
    }

    push(data){
        let newNode=new Node(data)
    if (this.top==null) {
        this.top=newNode;
    }
    else{
        newNode.next=this.top
        this.top=newNode;
    }
    }

    pop(){
let current=this.top
this.top=this.top.next
current.next=null        
    }

    display(){
        let current=this.top
        while (current!=null) {
            console.log(current.data);
            current=current.next;
        }
    }
}

const list=new stack()
list.push(2)
list.push(5)
list.push(7)
list.push(4)
list.display()
list.pop()
console.log(" ");
list.display()